moveit_ros/moveit_servo/src/servo_calcs.cpp

```diff
@@ -862,6 +861,13 @@ void ServoCalcs::suddenHalt(trajectory_msgs::msg::JointTrajectory& joint_traject
   }
 }
 
+void ServoCalcs::suddenHalt(sensor_msgs::msg::JointState& joint_state) const
+{
+  // Set the position to the original position, and velocity to 0
+  joint_state.position = original_joint_state_.position;
+  joint_state.velocity.assign(joint_state.position.size(), 0.0);
```

Comment:
Is joint_state.velocity.size() always the size of the joints? (i.e. do we ever not provide joint_state.velocity values? If so should we check that before populating the velocity fields here?)

Reply:
The `original_joint_state_` [contains only active joint](https://github.com/ros-planning/moveit2/blob/eb2c8b97862e9acb760dd0c7882d06d04e4ecfae/moveit_ros/moveit_servo/include/moveit_servo/servo_calcs.h#L284-L285) so this should return a joint state that only includes the active joints, which is what the rest of the calculations are expecting

Reply:
I think a better phrased question is "will incoming_joint_state_ always have velocity information"?

e.g. is there a servo configuration where `joint_state.velocity` would be empty here and we may be introducing side effects in ros control by resizing velocity?

Reply:
I can't think of any weird edge cases where the positions vector and the velocities vector would be different sizes (other than if one of them is completely empty).

Also, `internal_joint_state_` previously had the velocities filled ALWAYS while internal calculations are happening. So this looks OK to me. (https://github.com/ros-planning/moveit2/blob/eb2c8b97862e9acb760dd0c7882d06d04e4ecfae/moveit_ros/moveit_servo/src/servo_calcs.cpp#L158)

Velocities get dropped at L673 before the final message goes out, if configured to do so:

>   if (parameters_->publish_joint_velocities)
>     point.velocities = joint_state.velocity;